### PR TITLE
Disc 243/x button on search

### PR DESCRIPTION
**Pull request by jesperlauridsen**
This adds a X-button in the search-field that shows whenever a search has been done or something is in the queryfield.
Pressing the x resets to the frontpage with the portal stuff active.
Did some tinkering this the typing aswell.
Did some tinkering with the transitions aswell.

### CODE REVIEW

**Review by jorntx — COMMENTED** [review]
Looks good and tested out OK. Had some trouble with the backend-service fetching records so couldn't test it to the end.

Left some small comments. Look it over, respond and then we are good to go.

File: src/components/search/SearchWrapper.vue [hunk 1]
```diff
@@ -17,6 +21,32 @@ export default defineComponent({
...
+				searchResultStore.searchResult.length !== 0 ||
+				searchResultStore.searchFired === true
+			) {
+				xReset.value = true;
+			} else {
+				xReset.value = false;
+			}
+		});
+
+		watch(searchResultStore, (newValue, oldValue) => {
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/search/SearchWrapper.vue:39]: Good point. Will fix.

File: src/components/search/SearchWrapper.vue [hunk 2]
```diff
@@ -17,6 +21,32 @@ export default defineComponent({
...
+			if (
+				searchQuery.value.length !== 0 ||
+				newValue.searchResult.length !== 0 ||
+				searchResultStore.searchFired === true
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/search/SearchWrapper.vue:43]: Yea. I do this a lot - will try to be better.

File: src/components/search/wc-facet-checkbox.ts [hunk 3]
```diff
@@ -83,12 +83,12 @@ class checkboxComponent extends HTMLElement {
...
-			if (newValue === 'true') {
-				this.style.opacity = '1';
-				this.style.transform = 'translateY(0px)';
-			} else {
+			if (newValue === 'false') {
 				this.style.opacity = '0';
 				this.style.transform = 'translateY(-20px)';
+			} else {
+				/* this.style.opacity = '0';
+				this.style.transform = 'translateY(-20px)'; */
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/search/wc-facet-checkbox.ts:91]: Will remove it, it's was a leftover from previous times.

File: src/components/search/wc-searchbar.ts [hunk 4]
```diff
@@ -1,11 +1,11 @@
 class SearchComponent extends HTMLElement {
 	shadow: ShadowRoot;
-	query: string;
 	background: string | undefined;
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/search/wc-searchbar.ts:3]: Yea, will fix

File: src/components/search/wc-searchbar.ts [hunk 5]
```diff
@@ -24,6 +24,38 @@ class SearchComponent extends HTMLElement {
...
+
+	connectedCallback() {
+		const resetVal = this.getAttribute('reset-value');
+
+		const bgContainer = this.shadow.querySelector('.search-container') as HTMLElement;
+		if (bgContainer && this.background) {
+			bgContainer.style.backgroundImage = `url(${this.background})`;
+		}
+		if (resetVal) {
+			this.showXButton = JSON.parse(resetVal.toLowerCase());
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/search/wc-searchbar.ts:49]: Hmmm. That's true. I guess this is one of he downsides with passing stuff through to the web components like this, without it being specified. I'll let it be for now, but I think you're right.

File: src/components/search/wc-searchbar.ts [hunk 6]
```diff
@@ -24,6 +24,38 @@ class SearchComponent extends HTMLElement {
...
+			this.setResetVisibility(this.showXButton);
+		}
+	}
+
+	attributeChangedCallback(name: string, oldValue: string, newValue: string) {
+		if (name === 'reset-value') {
+			this.showXButton = JSON.parse(newValue.toLowerCase());
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/search/wc-searchbar.ts:56]: Same here, yes.

File: src/components/search/wc-searchbar.ts [hunk 7]
```diff
@@ -1,11 +1,11 @@
 class SearchComponent extends HTMLElement {
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/search/wc-searchbar.ts:1]: SearchBarComponent it is!

**Review by jorntx — COMMENTED** [review]
Code looks good now - thanks for the changes.

I gave it a spin and something seems to happen when you perform a search go to full record and then go back to the search again. The search is performed and the query is in the srach bar but now there is no way to reset your search. Can you reproduce?

**Comment by jesperlauridsen** [comment]
I could reproduce, and I've fixed it. Turns out we gotta watch the whole store to also catch the searchFired event in the watcher. Otherwise it'll fly under the radar on the mount.

**Review by jorntx — APPROVED** [review]
Yes - works now beautifully now.

Resolve the merge conflicts and give me a holla and I will merge:)

**Pull request merged**